### PR TITLE
TimeSeriesForest Classifier Fix

### LINKED ISSUE #1536: [BUG] TypeError for ComposableTimeSeriesForestClassifier
I was trying to follow the tutorial for "Univariate time series classification with sktime" but ran into an error with the time series forest classifier `ComposableTimeSeriesForestClassifier`. Copied the code exactly, everything was fine for a single time series tree. 

Code: 
```
tsf = ComposableTimeSeriesForestClassifier(
    estimator=time_series_tree,
    n_estimators=100,
    criterion="entropy",
    bootstrap=True,
    oob_score=True,
    random_state=1,
    n_jobs=-1,
)
```

Error: 
```
TypeError                                 Traceback (most recent call last)
<ipython-input-25-a9421957ffb2> in <module>()
      7     oob_score=True,
      8     random_state=1,
----> 9     n_jobs=-1,
     10 )

TypeError: Can't instantiate abstract class ComposableTimeSeriesForestClassifier with abstract methods _set_oob_score_and_attributes
```

(Google Colab, no GPU/TPU)

### PR DESCRIPTION
**Pull request by OliverMatthews**
#### Reference Issues/PRs

Fixes #1536 

#### What does this implement/fix? Explain your changes.

Fixed a bug preventing usage of ComposableTimeSeriesForestClassifier.

#### Does your contribution introduce a new dependency? If yes, which one?

No new dependencies.

#### What should a reviewer concentrate their feedback on?

An abstract method in sklearn called _set_oob_score_and_attributes was implemented to allow instantiation of ComposableTimeSeriesForestClassifier objects, but with just `pass` in the body of the method. This might need to be implemented properly in the future, but isn't currently used.

#### Any other comments?

None.

##### For all contributions
- [X] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [X] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by OliverMatthews** [comment]
@TonyBagnall

**Review by TonyBagnall — DISMISSED** [review]
Looks good, nice job

**Comment by TonyBagnall** [comment]
hi oli, its normal to link the issue that this PR is solving in the description. 
e..g. Fixed a bug preventing usage of ComposableTimeSeriesForestClassifier.

solves https://github.com/alan-turing-institute/sktime/issues/1536

**Comment by TonyBagnall** [comment]
Ah soz, shouldn't post in the evening!

**Comment by MatthewMiddlehurst** [comment]
Is the estimator_params change necessary? Not mentioned in the description.

File: sktime/classification/compose/_ensemble.py [hunk 1]
```diff
@@ -314,16 +308,19 @@ def _validate_estimator(self):
     def predict(self, X):
         """
         Predict class for X.
```
Comment (chrisholder) [inline sktime/classification/compose/_ensemble.py:310]: Move this comment up a line so:
`
"""Predict class for X.

File: sktime/classification/compose/_ensemble.py [hunk 2]
```diff
@@ -350,15 +347,18 @@ def predict(self, X):
     def predict_log_proba(self, X):
         """
         Predict class log-probabilities for X.
```
Comment (chrisholder) [inline sktime/classification/compose/_ensemble.py:349]: """Predict class log-probabilities for X.

File: sktime/classification/compose/_ensemble.py [hunk 3]
```diff
@@ -460,6 +462,10 @@ def _set_oob_score(self, X, y):
 
         self.oob_score_ = oob_score / self.n_outputs_
 
+    # TODO - Implement this abstract method properly.
+    def _set_oob_score_and_attributes(self, X, y):
+        pass
```
Comment (chrisholder) [inline sktime/classification/compose/_ensemble.py:467]: raise NotImplementedError("Not implemented")

**Review by chrisholder — CHANGES_REQUESTED** [review]
Looks good just raising an error and docstring formatting!

**Comment by OliverMatthews** [comment]
> Is the estimator_params change necessary? Not mentioned in the description.

Yes - without those two lines removed an AttributeError gets thrown as the ComposableTimeSeriesForestClassifier class no longer has an attribute for criterion, or a ValueError is thrown as the default parameter for min_impurity_split is invalid, and it's not used for anything else as far as I can tell

File: sktime/classification/compose/_ensemble.py (hunk 1 above)
Comment (OliverMatthews) [reply sktime/classification/compose/_ensemble.py]: Done 👍

File: sktime/classification/compose/_ensemble.py (hunk 2 above)
Comment (OliverMatthews) [reply sktime/classification/compose/_ensemble.py]: Done 👍

File: sktime/classification/compose/_ensemble.py (hunk 3 above)
Comment (OliverMatthews) [reply sktime/classification/compose/_ensemble.py]: Done 👍

**Comment by MatthewMiddlehurst** [comment]
None of the tests in other PRs raise an issue with the current version. Was this introduced with any other changes? Is your local package versions consistent with those used in sktime?

If it is a consequence of some other change thats fine. `min_impurity_split=None` is still a parameter in your changes if that is the case.

**Comment by MatthewMiddlehurst** [comment]
As far as i understand that list is used to set the parameters for the DecisionTreeClassifier base estimator. Has sklearn changed it and we allow that version?

**Review by chrisholder — APPROVED** [review]

**Comment by TonyBagnall** [comment]
I dont think this would have been picked up by existing tests since they do not test parameter tuning or oob calculation. I have had a nose around and I think this is a scikit learn problem as Oliver suggests, introduced in 0.24 I guess. I'm not sure how or even if the criterion gets set in the ensemble member. However, given this is currently broken and I would rather deprecate this class all together, I think I will approve this to hotfix the issue. This class will need revisiting when we sort out TSF business anyway.

**Review by MatthewMiddlehurst — DISMISSED** [review]

**Comment by OliverMatthews** [comment]
As Tony indicated, the existing tests don't test this so wouldn't have picked it up. I will take a look though at having left `min_impurity_split=None` as that probably ought to be removed as well, thanks for the catch @MatthewMiddlehurst

**Review by TonyBagnall — APPROVED** [review]

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**